pkg/controllers/csi/driver/server.go

```diff
@@ -223,6 +224,18 @@ func (srv *Server) unmount(volumeInfo csivolumes.VolumeInfo) {
 		}
 
 		for _, path := range needsCleanUp {
+			podInfoSymlinkPath := srv.findPodInfoSymlink(volumeInfo) // cleaning up the pod-info symlink here is far more efficient instead of having to walk the whole fs during cleanup
+			if podInfoSymlinkPath != "" {
+				_ = srv.fs.Remove(podInfoSymlinkPath)
+
+				podInfoSymlinkDir := filepath.Dir(podInfoSymlinkPath)
+
+				isEmpty, _ := srv.fs.IsEmpty(podInfoSymlinkDir)
```

Comment:
```suggestion
				if isEmpty, _ := srv.fs.IsEmpty(podInfoSymlinkDir); isEmpty {
				    _ = srv.fs.Remove(podInfoSymlinkDir)
				}
```
you just need it once, so worth to combine IMO together.

Reply:
done